### DISCUSSION TITLE
Questions about channel port numbers and ”Base Context Path“

### DISCUSSION THREAD

**Discussion by FeiTian01** [nextgenhealthcare/connect#4912] [Q&A] [answered]
My source is HTTPListener, he is only one "Base Context Path", and take up a port number, how can I use this "Base Context Path" to define different method name and to request  different destination

**Comment by pacmano1** [comment, marked as answer]
You can set your base context to whatever you want, let's say `/api`

It is up to you to parse everything after `/api` in your source connector to then make decision on what to do.   Look at the message mapping after sending a few message through - it will be obvious what variables you need to parse.

For example post to `/api/test`, `/apy/test?this=that` and so on.

**Reply by FeiTian01** [reply]
yes,i can parse request url, but how can I select different destinations according to different request url.
such as  I'm requesting url is 'testA', and I will send the data to destination A, or I'm requesting url is 'testB', and I will send the data to destination B,

**Reply by pacmano1** [reply]
Parse the URL and use destinatonSet filtering in a transformer.

http://javadocs.mirthcorp.com/connect/3.12.0/user-api/com/mirth/connect/server/userutil/DestinationSet.html

**Reply by FeiTian01** [reply]
thanks a lot !
